src/Microsoft.Health.Fhir.Core/Features/Search/SearchResultFilter.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Search.Filters;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    public sealed class SearchResultFilter : ISearchResultFilter
+    {
+        private readonly bool _isUSCoreEnabled;
+        private readonly bool _isSmartUserRequest;
+
+        public SearchResultFilter(IOptions<ImplementationGuidesConfiguration> implementationGuides, RequestContextAccessor<IFhirRequestContext> fhirRequestContextAccessor)
+        {
+            _isUSCoreEnabled = implementationGuides?.Value?.USCore?.MissingData ?? false;
```

Comment:
This class, SearchResultFilter seems to be designed to be expanded in the future to more search filters, if necessary.  Meaning that it is a container for all search filters.  However, it has detailed knowledge of the necessary parameters for MissingDataFilterCriteria.  If I were to add another filter, it seems I would need to update the constructor of this class, SearchResultFilter, to also have new input parameters for my new search filter.  Perhaps we could move the imlementationGuide and RequestContextAccessor to the MissingDataFilterCriteria constructor?  That way SearchResultFilter doesn't have to know about them.  Just an optional tought.

Reply:
Thanks for the suggestion. Let work on it  :)